flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSource.java

```diff
@@ -112,6 +113,11 @@ public boolean isBounded() {
 		return returnType;
 	}
 
+	@Override
+	public DataType getProducedDataType() {
+		return schema.toRowDataType();
```

Comment:
This is wrong fix, you should modify `returnType`.

Reply:
updated